what4/src/What4/Expr/Builder.hs

```diff
@@ -1101,8 +1148,95 @@ sbConcreteLookup sym arr0 mcidx idx
     -- Create select index.
   | otherwise = do
     case exprType arr0 of
-      BaseArrayRepr _ range ->
-        sbMakeExpr sym (SelectArray range arr0 idx)
+      BaseArrayRepr _ range -> do
+        (sliced_arr, sliced_idx) <- sliceArrayLookupUpdate sym arr0 idx
```

Comment:
Are you convinced this slicing stuff is worth it?  It breaks sharing pretty badly and is quite complicated, so unless we have evidence that it is really worthwhile in actual usage, I'd be inclined to leave it out.